src/charm.py

```diff
@@ -359,7 +359,7 @@ def __init__(self, *args):
         observe(self.on.show_external_endpoints_action, self._on_show_external_endpoints)  # type: ignore
 
         # Hook hollistic method
-        observe(self.on.traefik_pebble_ready, self.cleanup_tls_configuration)  # type: ignore
+        observe(self.on.traefik_pebble_ready, self.cleanup_tls_configuration)
         observe(self.on.start, self.cleanup_tls_configuration)
         observe(self.on.update_status, self.cleanup_tls_configuration)
         observe(self.on.config_changed, self.cleanup_tls_configuration)
```

Comment:
Do you need to run this on update status too? Also, isn't start always preceded by config changed?